apps/api/src/app/events/e2e/digest-events.e2e.ts

```diff
@@ -812,14 +826,110 @@ describe('Trigger event - Digest triggered events - /v1/events/trigger (POST)',
       _templateId: template._id,
       _subscriberId: subscriber._id,
     });
-
-    expect(messages[0].content).to.include(digests[0].payload.postId);
     expect(messages[1].content).to.include(digests[1].payload.postId);
+    expect(messages[2].content).to.include(digests[2].payload.postId);
+    const jobCount = await jobRepository.count({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+    });
+    const allJobs = await jobRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+    });
+    expect(jobCount).to.equal(11);
+  });
+
+  it('should create multiple digests based on different nested digestKeys', async function () {
+    const postId = MessageRepository.createObjectId();
+    const postId2 = MessageRepository.createObjectId();
+
+    template = await session.createTemplate({
+      steps: [
+        {
+          type: StepTypeEnum.DIGEST,
+          content: '',
+          metadata: {
+            unit: DigestUnitEnum.MINUTES,
+            amount: 5000,
+            digestKey: 'nested.postId',
+            type: DigestTypeEnum.REGULAR,
+          },
+        },
+        {
+          type: StepTypeEnum.CHAT,
+          content: 'Hello world {{nested.postId}}' as string,
+        },
+      ],
+    });
+    await triggerEvent({
+      customVar: 'No digest key',
+    });
+
+    await triggerEvent({
+      customVar: 'digest key1',
+      nested: { postId: postId },
+    });
+    await triggerEvent({
+      customVar: 'digest key2',
+      nested: { postId: postId2 },
+    });
+    await session.awaitParsingEvents();
+
+    await triggerEvent({
+      customVar: 'No digest key repeat',
+    });
+    await triggerEvent({
+      customVar: 'digest key1 repeat',
+      nested: { postId: postId },
+    });
+    await session.awaitParsingEvents();
+
+    await session.awaitRunningJobs(template?._id, false, 3);
+
+    let digests = await jobRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+      type: StepTypeEnum.DIGEST,
+    });
+    expect(digests[1].payload.nested.postId).not.to.equal(digests[2].payload.nested.postId);
+    expect(digests[0].payload.nested?.postId).to.equal(undefined);
+
+    expect(digests.length).to.equal(3);
+
+    digests = await jobRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+      type: StepTypeEnum.DIGEST,
+    });
+    for (const digest of digests) {
+      await runJob.execute(
+        RunJobCommand.create({
+          jobId: digest._id,
+          environmentId: digest._environmentId,
+          organizationId: digest._organizationId,
+          userId: digest._userId,
+        })
+      );
+    }
+
+    await session.awaitRunningJobs(template?._id, false, 0);
```

Comment:
You should be able to use the same solution as in the previous test you modified and just use the `awaitRunningJobs` testing utility.

```suggestion
    await session.awaitRunningJobs(template?._id, false, 3);
```
3 as you are expecting to have 3 digest jobs unfinished.

Reply:
@p-fernandez all the places these digest/delayed jobs we are triggering manually, so we can test the cases what happens after the digest test cases executes(after delay), we cant wait jobs to finish, as we have 10 seconds time out and test cases failing with timeout, so triggering all 3 jobs manually and waiting for all jobs to finish, so zero. make sense?